### DISCUSSION TITLE
The exe program runs for a while and then suddenly stops

### DISCUSSION THREAD

**Discussion by shadowmimosa** [pyinstaller/pyinstaller#6093] [PyInstaller]
The exe program runs for a while and then suddenly stops, and the resource usage is close to zero

How do I fix it or how do I troubleshoot it?

**Comment by rokm** [comment]
Is this a `noconsole`  / `windowed` build? What version of PyInstaller are you using?

**Reply by shadowmimosa** [reply]
> Is this a `noconsole` / `windowed` build? What version of PyInstaller are you using?

python == 3.8
pyinstaller == 4.2

My profile:
```
# -*- mode: python ; coding: utf-8 -*-
block_cipher = pyi_crypto.PyiBlockCipher(key='***')

a = Analysis(
    ['**.py'],
    pathex=[
        '**\\**\\**'
    ],
    binaries=[],
    datas=[],
    hiddenimports=[],
    hookspath=[],
    runtime_hooks=[],
    excludes=[],
    win_no_prefer_redirects=False,
    win_private_assemblies=False,
    cipher=block_cipher,
    noarchive=False)

pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)

exe = EXE(pyz,
          a.scripts,
          a.binaries,
          a.zipfiles,
          a.datas, [],
          name='**',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          upx_exclude=[],
          runtime_tmpdir=None,
          console=True,
          icon='*.ico')
```

**Reply by rokm** [reply]
>           console=True,

If you have console, then you can debug the problem by adding `print()` or `logger.info()` statements to your code to determine your program's flow and to find out where it crashed.

**Reply by shadowmimosa** [reply]
> > ```
> >       console=True,
> > ```
> 
> If you have console, then you can debug the problem by adding `print()` or `logger.info()` statements to your code to determine your program's flow and to find out where it crashed.

It is a circular program executed every minute.

In fact, the program didn't crash and stopped somewhere.
I tried to log every function and even catch exceptions in the main function. No errors were caught.

**Reply by rokm** [reply]
Hmm... Try setting `debug=True` in `EXE()`, which will print bootloader's debug messages.

As this seems to be `onefile` build; is this supposed to be long-running program? How long does it run before it finishes? Could some other service be removing its temporary directory while it is running? (You could try making a onedir build instead and see if that exhibits same problems; or try setting `runtime_tmpdir` to have onefile build unpack itself into a specific location outside the regular temporary directory).

**Reply by shadowmimosa** [reply]
> Hmm... Try setting `debug=True` in `EXE()`, which will print bootloader's debug messages.
> 
> As this seems to be `onefile` build; is this supposed to be long-running program? How long does it run before it finishes? Could some other service be removing its temporary directory while it is running? (You could try making a onedir build instead and see if that exhibits same problems; or try setting `runtime_tmpdir` to have onefile build unpack itself into a specific location outside the regular temporary directory).

thx. I'll try and test it.

**Reply by Ender29** [reply]
@shadowmimosa 
I have a same qustion，After setting upx=false, the problem has been solved
I guess some error when use upx pack some libs
